servers/quarkus-cli/src/intTest/java/org/projectnessie/quarkus/cli/BaseContentPersistTest.java

```diff
@@ -0,0 +1,150 @@
+import static org.projectnessie.versioned.storage.common.logic.Logics.commitLogic;
+import static org.projectnessie.versioned.storage.common.logic.Logics.referenceLogic;
+import static org.projectnessie.versioned.storage.common.objtypes.CommitHeaders.EMPTY_COMMIT_HEADERS;
+import static org.projectnessie.versioned.storage.common.objtypes.ContentValueObj.contentValue;
+import static org.projectnessie.versioned.storage.versionstore.TypeMapping.keyToStoreKey;
+import static org.projectnessie.versioned.store.DefaultStoreWorker.payloadForContent;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.quarkus.test.junit.main.LaunchResult;
+import io.quarkus.test.junit.main.QuarkusMainLauncher;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.io.TempDir;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.Namespace;
+import org.projectnessie.nessie.relocated.protobuf.ByteString;
+import org.projectnessie.versioned.storage.common.logic.CommitLogic;
+import org.projectnessie.versioned.storage.common.logic.CreateCommit.Builder;
+import org.projectnessie.versioned.storage.common.objtypes.CommitObj;
+import org.projectnessie.versioned.storage.common.objtypes.ContentValueObj;
+import org.projectnessie.versioned.storage.common.persist.Persist;
+import org.projectnessie.versioned.storage.common.persist.Reference;
+import org.projectnessie.versioned.store.DefaultStoreWorker;
+
+abstract class BaseContentPersistTest<OutputType> {
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private final Class<OutputType> outputClass;
+
+  @TempDir File tempDir;
+
+  protected LaunchResult result;
+  protected List<OutputType> entries;
+
+  private boolean testNamespaceCreated;
+
+  protected Namespace namespace;
+
+  BaseContentPersistTest(Class<OutputType> outputClass) {
+    this.outputClass = outputClass;
+  }
+
+  protected void launchNoFile(QuarkusMainLauncher launcher, String... args) {
+    launch(launcher, null, args);
+  }
+
+  protected void launch(QuarkusMainLauncher launcher, String... args) throws Exception {
+    File output = new File(tempDir, "check-content.json");
+    launch(launcher, output, args);
+    JavaType type = MAPPER.getTypeFactory().constructCollectionType(List.class, outputClass);
+    entries = MAPPER.readValue(output, type);
+  }
+
+  private void launch(QuarkusMainLauncher launcher, File outputFile, String... args) {
+    List<String> cmdArgs = new ArrayList<>(Arrays.asList(args));
+
+    if (outputFile != null) {
+      cmdArgs.add("--output");
+      cmdArgs.add(outputFile.getAbsolutePath());
+    }
+
+    result = launcher.launch(cmdArgs.toArray(new String[0]));
+  }
```

Comment:
Would it be too much trouble to factor this out into a common class wrt `BaseContentTest`?

Reply:
It was a bit involved but I managed to remove all duplications using `@Nested` tests.